src/Spectre.libClassifier/RandomSplitter.cpp

```diff
@@ -0,0 +1,72 @@
+﻿/*
+* RandomSplitter.cpp
+* Splits dataset into training and control datasets.
+*
+Copyright 2017 Grzegorz Mrukwa, Wojciech Wilgierz
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "RandomSplitter.h"
+#include <random>
+#include "Spectre.libPlatform/Filter.h"
+
+namespace Spectre::libClassifier {
+
+RandomSplitter::RandomSplitter(double trainingPercent, Seed rngSeed)
+    : m_trainingPercent(trainingPercent),
+      m_randomNumberGenerator(rngSeed)
+{
+
+}
+
+std::pair<Spectre::libClassifier::OpenCvDataset, Spectre::libClassifier::OpenCvDataset> RandomSplitter::split(const Spectre::libClassifier::OpenCvDataset& data)
+{
+    std::vector<bool> flags;
+    std::bernoulli_distribution dist(m_trainingPercent);
+
+    for (auto i = 0u; i < data.size(); i++)
+    {
+        flags.push_back(dist(m_randomNumberGenerator));
```

Comment:
It would be beneficial for all of us, if something like this will be extracted as a common pattern to `libPlatform`. What I mean is construction of `std::vector` of specified size, but with elements specified by an execution of a lambda.